VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+    $sddcMgr = Get-VCFManager
+    $sddcMgrVersion = $sddcMgr.version.split(".")[0]
+
+    $resourcesObject = @()
+    # SDDC Manager
+    IF ($domainType -eq "Management")
+    {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $sddcMgr.fqdn
+            'name'       = $sddcMgr.fqdn.split(".")[0]
+            'resourceId' = $sddcMgr.id
+            'type'       = "SDDC_MANAGER"
+        } 
+    }
+
+    # vCenters
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq "MGMT" }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    elseIf (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Workload"))
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+        $isLegacyWldnsxT = Get-VCFNsxtCluster | Where-Object {$_.domains.id -eq $domain.id}
+        $isLegacyWldnsxV = Get-VCFNsxvManager | Where-Object {$_.domain.id -eq $domain.id}
+    }
+    elseIf (([float]$sddcMgrVersion -ge [float]4) -AND ($domainType -eq "Management"))
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.type -eq "MANAGEMENT" }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    else
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    Foreach ($vCenter in $vCenterServer) {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $vCenter.fqdn
+            'name'       = $vCenter.fqdn.split(".")[0]
+            'resourceId' = $vCenter.id
+            'type'       = "VCENTER"
+        }
+    }
+
+    # NSX V Manager
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management") -OR ($isLegacyWldnsxV))
+    {
+        $nsxvManager = Get-VCFNsxvManager | Where-Object { $_.domain.id -eq $domain.id }
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $nsxvManager.fqdn
+            'name'       = $nsxvManager.fqdn.split(".")[0]
+            'resourceId' = $nsxvManager.id
+            'type'       = "NSX_MANAGER"
+        }
+    }
+
+    #VRLI
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $logInsight = Get-VCFvRLI *>$null
+        If ($logInsight)
+        {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = ($logInsight.nodes | Where-Object {$_.type -eq "MASTER"}).fqdn
+                'name'       = (($logInsight.nodes | Where-Object {$_.type -eq "MASTER"}).fqdn).split(".")[0]
+                'resourceId' = $logInsight.id
+                'type'       = "VRLI"
+            }    
+        }
+    }
+    
+    #PSCs
```

Comment:
```suggestion
    # Platform Services Controllers (Embedded)
```

Reply:
section removed due to reference to VCF 3.x
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)